src/mochiweb_http.erl

```diff
@@ -168,8 +174,14 @@ collect_headers(Socket, Request, Body, Collected, Trunc, HeaderCount) ->
         {ssl_closed, _} ->
             mochiweb_socket:close(Socket),
             exit(normal);
-        _Other ->
-            handle_invalid_request(Socket, Request, [])
+        {ProtocolErr, _Socket, _Reason} when ProtocolErr =:= tcp_error orelse ProtocolErr =:= ssl_error ->
```

Comment:
We could add a check that the message received is related to the current socket by matching on the Socket in the receive statement (should really be done for all the receive clauses). 

+1 for close_on_error - that makes the error situation stand out and makes it easier to trace. It is merely adding a very small function, so it is not that big of a change.